clients/pubsub-common/src/test/java/org/eclipse/hono/client/pubsub/AbstractPubSubBasedMessageSenderTest.java

```diff
@@ -0,0 +1,222 @@
+/**
+ * Copyright (c) 2022 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.hono.client.pubsub;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import java.net.HttpURLConnection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+
+import org.eclipse.hono.client.ClientErrorException;
+import org.eclipse.hono.client.ServerErrorException;
+import org.eclipse.hono.util.LifecycleStatus;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+
+import com.google.protobuf.ByteString;
+import com.google.pubsub.v1.PubsubMessage;
+
+import io.opentracing.Tracer;
+import io.opentracing.noop.NoopSpan;
+import io.opentracing.noop.NoopTracerFactory;
+import io.vertx.core.Future;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.junit5.VertxTestContext;
+
+/**
+ * Verifies the generic behavior of {@link AbstractPubSubBasedMessageSender}.
+ */
+public class AbstractPubSubBasedMessageSenderTest {
+
+    private final String TOPIC = "event";
+
+    private final String PROJECT_ID = "test-project";
+
+    private final String TENANT_ID = "tenant";
+
+    private final String DEVICE_ID = "device";
+
+    private final Tracer tracer = NoopTracerFactory.create();
+
+    private LifecycleStatus lifecycleStatus;
+
+    private CachingPubSubPublisherFactory factory;
+
+    private AbstractPubSubBasedMessageSender sender;
+
+    /**
+     * Sets up the fixture.
+     */
+    @BeforeEach
+    public void setUp() {
+        factory = mock(CachingPubSubPublisherFactory.class);
```

Comment:
IMHO the AbstractPubSubBasedMessageSender does not depend on the concrete CachingPubSubPublisherFactory to be used as the factory, or am I mistaken?
If not, then FMPOV we should mock the *PubSubPublisherFactory* interface only.

Reply:
You are right, mocked the _PubSubPublisherFactory_ interface